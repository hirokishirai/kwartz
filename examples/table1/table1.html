<?xml version="1.0"?>
<html>
  <body>

    <table>
      <tr bgcolor="#CCCCFF" id="mark:row">
        <td id="mark:name">Foo</td>
        <td id="mark:mail">foo@mail.com</td>
      </tr>
      <tr bgcolor="#FFCCCC" id="dummy:d1">
        <td>Bar</td>
        <td>bar@mail.net</td>
      </tr>
      <tr bgcolor="#CCCCFF" id="dummy:d2">
        <td>Baz</td>
        <td>baz@mail.org</td>
      </tr>
    </table>

  </body>
</html>
