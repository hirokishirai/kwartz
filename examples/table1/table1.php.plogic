#row {
  attrs:  "bgcolor" $color;
  logic: {
    $i = 0;
    foreach ($list as $item) {
      $color = ++$i % 2 == 0 ? '#FFCCCC' : '#CCCCFF';
      _elem();
    }
  }
}

#name {
  value:  $item['name'];
}

#mail {
  value:  $item['mail'];
}

#dummy1 {
  logic: { }   /* ignore this element */
}
