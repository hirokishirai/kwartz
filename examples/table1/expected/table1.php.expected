<html>
  <body>
  
    <table>
      <tr bgcolor="#CCCCFF">
        <td>Nomo</td>
        <td>hideo@mail.com</td>
      </tr>
      <tr bgcolor="#FFCCCC">
        <td>Ichiro</td>
        <td>ichiro@mail.net</td>
      </tr>
      <tr bgcolor="#CCCCFF">
        <td>Matsui</td>
        <td>hideki@mail.org</td>
      </tr>
    </table>

 </body>
</html>
